### PR TITLE
Publication fixes, security updates, and hotfixes

### PR DESCRIPTION
**Pull request by cjsauer**
### What I did:

+ Moved getUrls into a server only method, because it depends on an NPM package that won't be available to the client
+ Fixed all of the channel publication tests. These were some pretty big changes. The tests used to be giving false positives. They now fail correctly. (And hopefully pass correctly)
+ Added a blank route to the fixtures package. This makes the testing of publications much easier, as there won't be any conflicting publications on the page already.
+ Deny any editing to the User's profile field. For some reason in Meteor it is the default to allow *any* editing to the user profile field.
+ Added a method for subscribing to channels. Now that the profile is not editable, we need a method to subscribe to channels from the client (added tests for this method too)

### CODE REVIEW

File: packages/fixtures/routes.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FlowRouter.route('/blank', {
+  action: function() {
+    // Do nothing...
```
Comment (elGatoMantocko) [inline packages/fixtures/routes.js]: Nice

**Comment by elGatoMantocko** [comment]
This is looking pretty good. I am going to merge it into the subscribe branch to just settle the conflicts now.

**Comment by elGatoMantocko** [comment]
So I see that there is a /users/subscribe method, should there also be a /users/unsubscribe method?

**Comment by cjsauer** [comment]
@elGatoMantocko yeah there definitely should be. I'll push that up once I've written and tested it.

**Comment by elGatoMantocko** [comment]
Should the User_spec.js file define tests for subscription and unsubscription now that those are in their own tests? Right now the tests are passing, but after merging into [subscribe](https://github.com/elGatoMantocko/BetterBackground/tree/subscribe) the 'can unsubscribe from channels' test started throwing errors.

**Comment by cjsauer** [comment]
The tests in User_spec are testing different things than the tests we just wrote for the methods. One is making sure that those collection functions are actually working correctly, the other are testing that those collection methods are actually *called* (and the user is logged in, etc.). We need both for full path coverage.

**Comment by elGatoMantocko** [comment]
Alrighty. I will see what I can do to make them work.

**Pull request closed without merging**